### PR TITLE
Return exit code 1 when buildah-rmi fails

### LINKED ISSUE #409: incorrect return code when buildah rmi fails
The return code should be 1.

```
[root@bootp-73-3-165 ~]# buildah from busybox
busybox-working-container
[root@bootp-73-3-165 ~]# buildah rmi busybox
Could not remove image "busybox" (must force) - container "ec808443580e9eef8e9ef1a69a63a2186d7c6cdb323f3e7adc0a8d0fd55038a6" is using its reference image: image is in use by a container
[root@bootp-73-3-165 ~]# echo $?
0
```

### PR DESCRIPTION
**Pull request by ripcurld0**
Fixes #409

### CODE REVIEW

**Comment by rhatdan** [comment]
Looks like your fix is actually showing a whole bunch of bugs in our testing.  Perhaps we have to change these build tests to do  `buildah rmi -f`  Or remove the container in a separate step then the rmi

**Comment by ripcurld0** [comment]
Luckily there are only 3 tests that fail.

File: tests/from.bats [hunk 1]
```diff
@@ -12,12 +12,10 @@ load helpers
 
   cid=$(buildah from --pull --signature-policy ${TESTSDIR}/policy.json dir:${elsewhere})
   buildah rm $cid
-  buildah rmi ${elsewhere}
```
Comment (rhatdan) [inline tests/from.bats]: Is it ok to not remove the elsewhere image?  Are we relying on the test cleanup to get rid of it?  Do we know why the buildah rmi was failing?

File: tests/from.bats [hunk 2]
```diff
@@ -12,12 +12,10 @@ load helpers
...
   [ "$cid" = elsewhere-img-working-container ]
 
   cid=$(buildah from --pull-always --signature-policy ${TESTSDIR}/policy.json dir:${elsewhere})
   buildah rm $cid
-  buildah rmi ${elsewhere}
```
Comment (rhatdan) [inline tests/from.bats]: Ditto

File: tests/from.bats (hunk 1 above)
Comment (ripcurld0) [reply tests/from.bats]: I assumed that when you're using an image from a directory (`dir:`) it is not saved in `/var/run/containers/storage` and thus no need to run `buildah rmi` upon it.

I might be wrong and that's fine. Let me know and I work it out. 😃

Comment (rhatdan) [reply tests/from.bats]: No I believe that it gets pulled from there and stored in containers/storage.  Could you confirm or deny my theory.  :^)

Comment (ripcurld0) [reply tests/from.bats]:
```
# export elsewhere=/tmp/image1
# mkdir -p ${elsewhere}
# tree ${elsewhere}
/tmp/image1

0 directories, 0 files
# buildah from --name cnt1 busybox
cnt1
# buildah commit cnt1 dir:${elsewhere}
Getting image source signatures
Copying blob sha256:57310166fe88e0dc63a80ca5c219283a932db0f3969712e2f8a86ada143bf566
 706.12 KiB / 706.12 KiB [==================================================] 0s
Copying blob sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1
 32 B / 32 B [==============================================================] 0s
Copying config sha256:6209ef68c57fd2e54be91dd3393b281dcac41035ee9648d67198cc779cd0803f
 711 B / 711 B [============================================================] 0s
Writing manifest to image destination
Storing signatures
# buildah images
IMAGE ID             IMAGE NAME                                               CREATED AT             SIZE
5b0d59026729         docker.io/library/busybox:latest                         Jan 24, 2018 04:29     1.299 MB
# tree ${elsewhere}
/tmp/image1
├── 4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1.tar
├── 57310166fe88e0dc63a80ca5c219283a932db0f3969712e2f8a86ada143bf566.tar
├── 6209ef68c57fd2e54be91dd3393b281dcac41035ee9648d67198cc779cd0803f.tar
├── manifest.json
└── version

0 directories, 5 files
```

@nalind @TomSweeneyRedHat can you confirm this?

File: tests/from.bats (hunk 2 above)
Comment (TomSweeneyRedHat) [reply tests/from.bats]: Actually removing the  ${elsewhere} image wasn't working based on my testing.   I think it was failing quietly.

```
# buildah rmi ${elsewhere}
could not get image "/tmp/image1": identifier is not an image
[root@localhost buildah]# buildah rmi dir:${elsewhere}
could not get image "dir:/tmp/image1": error reading image using reference "dir:/tmp/image1": image not known
# buildah images
IMAGE ID             IMAGE NAME                                               CREATED AT             SIZE
5b0d59026729         docker.io/library/busybox:latest                         Jan 24, 2018 04:29     1.299 MB
# buildah rmi 5b0d59026729
```

Comment (TomSweeneyRedHat) [reply tests/from.bats]: Ok, here's my full test.  The busybox image is getting stored in /var/lib/containers/storeage/overlay-images.  The commit is placing a copy of the image into /tmp/image1 and that is not being removed even with 'buildah rmi --all'.  I think that's OK, but am not 100% sure.

```
# export elsewhere=/tmp/image1
# mkdir -p ${elsewhere}
# tree ${elsewhere}
/tmp/image1

0 directories, 0 files
# tree /var/lib/containers/storage/overlay-images
/var/lib/containers/storage/overlay-images
├── images.json
└── images.lock

0 directories, 2 files
# buildah from --name cnt1 busybox
Getting image source signatures
Copying blob sha256:57310166fe88e0dc63a80ca5c219283a932db0f3969712e2f8a86ada143bf566
 706.12 KiB / 706.12 KiB [==================================================] 0s
Copying config sha256:5b0d59026729b68570d99bc4f3f7c31a2e4f2a5736435641565d93e7c25bd2c3
 1.46 KiB / 1.46 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
cnt1
# tree ${elsewhere}
/tmp/image1

0 directories, 0 files
# tree /var/lib/containers/storage/overlay-images
/var/lib/containers/storage/overlay-images
├── 5b0d59026729b68570d99bc4f3f7c31a2e4f2a5736435641565d93e7c25bd2c3
│   ├── =c2hhMjU2OjViMGQ1OTAyNjcyOWI2ODU3MGQ5OWJjNGYzZjdjMzFhMmU0ZjJhNTczNjQzNTY0MTU2NWQ5M2U3YzI1YmQyYzM=
│   └── manifest
├── images.json
└── images.lock

1 directory, 4 files

# buildah commit cnt1 dir:${elsewhere}
Getting image source signatures
Copying blob sha256:57310166fe88e0dc63a80ca5c219283a932db0f3969712e2f8a86ada143bf566
 706.12 KiB / 706.12 KiB [==================================================] 0s
Copying blob sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1
 32 B / 32 B [==============================================================] 0s
Copying config sha256:0730070e2a153d8e76a5aba3512fce65b2f1fdf176afc3b98281e13b612d45f7
 711 B / 711 B [============================================================] 0s
Writing manifest to image destination
Storing signatures

# tree ${elsewhere}
/tmp/image1
├── 0730070e2a153d8e76a5aba3512fce65b2f1fdf176afc3b98281e13b612d45f7.tar
├── 4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1.tar
├── 57310166fe88e0dc63a80ca5c219283a932db0f3969712e2f8a86ada143bf566.tar
├── manifest.json
└── version

0 directories, 5 files
# tree /var/lib/containers/storage/overlay-images
/var/lib/containers/storage/overlay-images
├── 5b0d59026729b68570d99bc4f3f7c31a2e4f2a5736435641565d93e7c25bd2c3
│   ├── =c2hhMjU2OjViMGQ1OTAyNjcyOWI2ODU3MGQ5OWJjNGYzZjdjMzFhMmU0ZjJhNTczNjQzNTY0MTU2NWQ5M2U3YzI1YmQyYzM=
│   └── manifest
├── images.json
└── images.lock

1 directory, 4 files

# buildah images
IMAGE ID             IMAGE NAME                                               CREATED AT             SIZE
5b0d59026729         docker.io/library/busybox:latest                         Jan 24, 2018 04:29     1.299 MB
# buildah containers
CONTAINER ID  BUILDER  IMAGE ID     IMAGE NAME                       CONTAINER NAME
f0588e8de0ff     *     5b0d59026729 docker.io/library/busybox:latest cnt1

# buildah rm cnt1
f0588e8de0ff7276ce4e0827740069e86c95413d8143b1a0b4c44b153c82f30a

# tree ${elsewhere}
/tmp/image1
├── 0730070e2a153d8e76a5aba3512fce65b2f1fdf176afc3b98281e13b612d45f7.tar
├── 4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1.tar
├── 57310166fe88e0dc63a80ca5c219283a932db0f3969712e2f8a86ada143bf566.tar
├── manifest.json
└── version

0 directories, 5 files
# tree /var/lib/containers/storage/overlay-images
/var/lib/containers/storage/overlay-images
├── 5b0d59026729b68570d99bc4f3f7c31a2e4f2a5736435641565d93e7c25bd2c3
│   ├── =c2hhMjU2OjViMGQ1OTAyNjcyOWI2ODU3MGQ5OWJjNGYzZjdjMzFhMmU0ZjJhNTczNjQzNTY0MTU2NWQ5M2U3YzI1YmQyYzM=
│   └── manifest
├── images.json
└── images.lock

1 directory, 4 files

# buildah rmi ${elsewhere}
could not get image "/tmp/image1": identifier is not an image
# buildah rmi dir:${elsewhere}
could not get image "dir:/tmp/image1": error reading image using reference "dir:/tmp/image1": image not known
# buildah images
IMAGE ID             IMAGE NAME                                               CREATED AT             SIZE
5b0d59026729         docker.io/library/busybox:latest                         Jan 24, 2018 04:29     1.299 MB
# buildah rmi 5b0d59026729
5b0d59026729b68570d99bc4f3f7c31a2e4f2a5736435641565d93e7c25bd2c3

# tree ${elsewhere}
/tmp/image1
├── 0730070e2a153d8e76a5aba3512fce65b2f1fdf176afc3b98281e13b612d45f7.tar
├── 4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1.tar
├── 57310166fe88e0dc63a80ca5c219283a932db0f3969712e2f8a86ada143bf566.tar
├── manifest.json
└── version

0 directories, 5 files
# tree /var/lib/containers/storage/overlay-images
/var/lib/containers/storage/overlay-images
├── images.json
└── images.lock

0 directories, 2 files

# buildah rmi --all
# buildah rm --all
# tree ${elsewhere}
/tmp/image1
├── 0730070e2a153d8e76a5aba3512fce65b2f1fdf176afc3b98281e13b612d45f7.tar
├── 4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1.tar
├── 57310166fe88e0dc63a80ca5c219283a932db0f3969712e2f8a86ada143bf566.tar
├── manifest.json
└── version
```

File: tests/from.bats [hunk 3]
```diff
@@ -26,12 +24,10 @@ load helpers
...
   [ "$cid" = `basename ${elsewhere}`-working-container ]
 }
```
Comment (TomSweeneyRedHat) [inline tests/from.bats]: Probably ought to add a 'rm -rf ${elsewhere}/* here at the end.   Might not hurt to add 'buildah rm --all' and 'buildah rmi --all' in that order too.

Comment (ripcurld0) [reply tests/from.bats]: Thanks. will do so.

**Comment by rhatdan** [comment]
I think the image created is not names $elsewhere, it must have a different name in storage, so we should be removing that name.

**Comment by rhatdan** [comment]
@ripcurld0 Needs a rebase.

**Comment by ripcurld0** [comment]
@rhatdan thanks!

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by ripcurld0** [comment]
LOL, I am working on this merge conflict.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by ripcurld0** [comment]
Thank you 🌈

**Pull request closed without merging**